sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -109,7 +109,7 @@ func Main(ctx context.Context, loggingEndpoint, controlEndpoint string) error {
 			log.Debugf(ctx, "RECV: %v", proto.MarshalTextString(req))
 			recordInstructionRequest(req)
 
-			hooks.RunRequestHooks(ctx, req)
+			ctx = hooks.RunRequestHooks(ctx, req)
```

Comment:
We should use a local variable for the returned context to limit the scope to just the present request/response.

Reply:
The ctx is now passed into the closure so its lifetime is properly scoped.